### PR TITLE
Fix bugs breaking wikipedia cats in chapter 15

### PR DESCRIPTION
**Pull request by chrishtr**
1. Protocol-relative URLs were not supported, and that's most of the images on this page.
2. The `AttributeParser` got confused in some situations where a style sheet ended up being parsed as an attribute. I fixed it by eating the rest of the string if `word` fails to advance.
3. A typo in the exception handler for failed image loads.

Now this URL can be loaded, and the images show: https://en.wikipedia.org/wiki/Cat

### CODE REVIEW

File: book/styles.md [hunk 1]
```diff
@@ -598,6 +598,9 @@ including query-relative and scheme-relative URLs, that I'm skipping.]
     existing scheme and host; or
 -   A path-relative URL, which doesn't start with a slash and is
     resolved like a file name would be.
+-   A protocol-relative URL that starts with "//" followed by a full URL,
+    which should be completed to a full url using the same protocol
+    (`http` or` https`) as the current URL.
```
Comment (pavpanchekha) [inline book/styles.md:603]:
```suggestion
    which should use the existing scheme.
```

Comment (pavpanchekha) [reply book/styles.md:603]: We call it a "scheme" throughout.

**Review by pavpanchekha — APPROVED** [review]
Great! I'm personally still a bit torn on `AttributeParser`. To my knowledge it does not appear in the book, so I think we should drop it entirely from the code. But either way it should work!

File: book/styles.md [hunk 2]
```diff
@@ -598,6 +598,8 @@ including query-relative and scheme-relative URLs, that I'm skipping.]
     existing scheme and host; or
 -   A path-relative URL, which doesn't start with a slash and is
     resolved like a file name would be.
+-   A protocol-relative URL that starts with "//" followed by a full URL,
```
Comment (chrishtr) [inline book/styles.md:601]:
```suggestion
-   A scheme-relative URL that starts with "//" followed by a full URL,
```

**Pull request merged**